File: frontend/src/api/parkings.ts
import { api } from './client';
import type { Parking, ParkingDetail } from '@/types/parking';

export const getParkings = () => api.get<Parking[]>('/parkings').then(r => r.data);
export const getParking = (id: number) => api.get<ParkingDetail>(`/parkings/${id}`).then(r => r.data);

File: backend/app/routers/parkings.py
from fastapi import APIRouter, HTTPException

from app.schemas.parking import Parking, ParkingDetail
from app.services.fleet import get_parking, list_parkings

router = APIRouter(prefix="/api/parkings", tags=["parkings"])


@router.get("", response_model=list[Parking])
async def list_parkings_route() -> list[Parking]:
    return list_parkings()


@router.get("/{parking_id}", response_model=ParkingDetail)
async def get_parking_route(parking_id: int) -> ParkingDetail:
    result = get_parking(parking_id)
    if not result:
        raise HTTPException(status_code=404, detail="Стоянка не найдена")
    return result
